cores/esp8266/PolledTimeout.h

```diff
@@ -47,15 +47,40 @@ struct YieldOrSkip
 
 } //YieldPolicy
 
+namespace TimePolicy
+{
 
-template <bool PeriodicT, typename YieldPolicyT = YieldPolicy::DoNothing>  
+struct TimeMillis
```

Comment:
The TimeMillis and TimeCycle policies have radically different rollover constraints. I think it merits implementing something like a static constexpr timeType maxTime(), or timeMax(), or just max(), and the timeoutTemplate constructor argument should be checked against it.
Up for discussion: if the constructor timeout argument is bigger than max, what should the behavior be?
Because it's an error condition on construction, the usual approach is to throw a C++ Exception, but we don't have them enabled by default yet, so we can't do that.
Maybe a debug msg and setting _timeout to 0 to force construction as expired?